app/src/main/java/org/schabi/newpipe/player/mediabrowser/MediaBrowserConnector.kt

```diff
@@ -0,0 +1,738 @@
+                    mediaId, playWhenReady, extras
+                )
+            )
+        }
+
+        disposePrepareOrPlayCommands()
+        prepareOrPlayDisposable = extractPlayQueueFromMediaId(mediaId)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(
+                { playQueue ->
+                    sessionConnector.setCustomErrorMessage(null)
+                    NavigationHelper.playOnBackgroundPlayer(
+                        playerService, playQueue,
+                        playWhenReady
+                    )
+                },
+                { throwable ->
+                    playbackError(
+                        ErrorInfo(
+                            throwable, UserAction.PLAY_STREAM,
+                            "Failed playback of media ID [$mediaId]: "
+                        )
+                    )
+                }
+            )
+    }
+
+    override fun onPrepareFromSearch(
+        query: String,
+        playWhenReady: Boolean,
+        extras: Bundle?
+    ) {
+        disposePrepareOrPlayCommands()
+        playbackError(
+            R.string.content_not_supported,
+            PlaybackStateCompat.ERROR_CODE_NOT_SUPPORTED
+        )
+    }
+
+    private fun searchMusicBySongTitle(query: String): Single<SearchInfo> {
+        val serviceId = ServiceHelper.getSelectedServiceId(playerService)
+        return ExtractorHelper.searchFor(
+            serviceId, query,
+            ArrayList(), ""
+        )
+    }
+
+    private fun mediaItemsFromInfoItemList(result: ListInfo<InfoItem>): SingleSource<List<MediaBrowserCompat.MediaItem>> {
+        result.errors
+            .takeIf { exceptions ->
+                exceptions.isNotEmpty() &&
+                    !(
+                        exceptions.size == 1 &&
+                            exceptions.first() is NothingFoundException
+                        )
+            }?.let { exceptions ->
+                return Single.error(exceptions.first())
+            }
+
+        val items = result.relatedItems
+        if (items.isEmpty()) {
+            return Single.error(NullPointerException("Got no search results."))
+        }
+        try {
+            val results =
+                items
+                    .filter { item ->
+                        item.infoType == InfoType.STREAM ||
+                            item.infoType == InfoType.PLAYLIST ||
+                            item.infoType == InfoType.CHANNEL
+                    }.map { item -> this.createInfoItemMediaItem(item) }
+
+            return Single.just(results)
+        } catch (error: Exception) {
+            return Single.error(error)
+        }
+    }
+
+    private fun handleSearchError(throwable: Throwable) {
+        Log.e(TAG, "Search error: $throwable")
+        disposePrepareOrPlayCommands()
+        playbackError(R.string.content_not_supported, PlaybackStateCompat.ERROR_CODE_NOT_SUPPORTED)
```

Comment:
Why is a content not supported error raised when a search error occurs?

Reply:
I wasn't sure how to translate the possible internal errors, so I just picked one. I suppose we could generate ERROR_CODE_ACTION_ABORTED or another value as a general error for search errors, or alternatively try to translate the individual potential errors. Do you know if the extractor has a list of potential exceptions?

Reply:
The error code has been updated to ERROR_CODE_APP_ERROR to reflect the issue in the search function.